elyra/util/cos.py

```diff
@@ -46,16 +46,23 @@ def __init__(self, config=None, endpoint=None, access_key=None, secret_key=None,
             self.bucket = config.metadata['cos_bucket']
             if auth_type in ['USER_CREDENTIALS', 'KUBERNETES_SECRET']:
                 # TODO remove temporary validation checks after
-                # the 'cos_username' and 'cos_password' schema properties
-                # are marked as required.
+                # metadata can properly enforce constraints
                 if len(config.metadata.get('cos_username', '').strip()) == 0 or\
                    len(config.metadata.get('cos_password', '').strip()) == 0:
-                    raise RuntimeError(f'Authentication provider \'{auth_type}\' requires '
+                    raise RuntimeError('Cannot connect to object storage. '
+                                       f'Authentication provider \'{auth_type}\' requires '
```

Comment:
I would suggest this be moved into the respective metadata class and let metadata properly enforce the constraints via the available "save hooks".